src/misc/materials.jl

```diff
@@ -0,0 +1,201 @@
+abstract type AbstractMaterials end
```

Comment:
I don't think this is giving you anything. Is there any circumstance under which you would want to use a generic "material" without knowing anything about it in more detail? There is no real relationship between the different concrete material types you've defined, so unless you have something in mind for this generic case, I don't think you need to derive them from a common abstract type like this.

Reply:
Agreed, I don't really have a use case for this now. Will get rid of it.